### DISCUSSION TITLE
Is this a downside of using @apply?

### DISCUSSION THREAD

**Discussion by abdennourzahaf** [tailwindlabs/tailwindcss#4745] [Help]
https://tailwindcss.com/docs/extracting-components#extracting-component-classes-with-apply
Using @apply results in the following HTML code:

```
<button class="btn-blue">Click me</button>
```

Instead of replacing the class name "btn-blue" with the applied tailwind class utilities, it includes the styles in a class with the name "btn-blue".

My question is, doesn't that take the atomic CSS benefits away?

**Comment by MartinHughes-BPC** [comment]
I remember Adam saying that @apply was really a trick to get people who didn't like the idea of lots of utility classes cluttering up the html to use Tailwind .

I tend not to use it as I find it easier (and more fun) to create reusable components.
